### ISSUE TITLE
Increase code coverage

### ISSUE THREAD

**Issue by dcorrea777** [Respect/Validation#1456]
Hey everyone,

I'm looking to get back into contributing to the project and I've been checking out the roadmap for version 3.0. I want to start by increasing the code coverage, but I ran into a few questions.

I've been adding some exception tests using phpt, but I got stuck on the Base rule test. This rule throws an exception called `InvalidRuleConstructorException`. However, it seems like this exception isn't being caught by the `exceptionMessage` and `exceptionFullMessage` functions. Am I doing something wrong?

**Comment by dcorrea777** [comment]
I think I've found the right way to do exception testing, I can follow phpunit itself, correct?

Example:

```php
#[Test]
public function itShouldThrowsExceptionWhenBaseIsNotValid(): void
{
    $this->expectException(InvalidRuleConstructorException::class);
    $this->expectExceptionMessage('a base between 1 and 62 is required');

    (new Base(63))->evaluate('011010001');
}
```

**Comment by henriquemoody** [comment]
Hey hey! 👋 

I've commented on #1457 that it's best not to include the call to the `evaluate()` method because it gives the impression that the exception is thrown from that method and not from the Rule's constructor.

**Comment by dcorrea777** [comment]
Thanks for the reply @henriquemoody
